documentation/codeQuality/README.md

```diff
@@ -1,2 +1,233 @@
+
+1. Go to build.gradle and then add JaCoCo into plugins section. Here is what our plugin section looks like.
+```
+plugins {
+    // Apply the java plugin to add support for Java
+    id 'java'
+
+    // Apply the application plugin to add support for building an application
+    id 'application'
+
+    id "org.sonarqube" version "2.6"
+
+    // Add this line
+    id "jacoco"
+}
+```
+
+2. To see the test report, run the following command
+```
+gradle test jacocoTestReport
+```
+
+3. By default, you test report will be located under .build/reports/jacoco
+
+4. Now let's return the SonarCloud analyze command. Your coverage result should be present on the SonarCloud's site.
+
+5. In Addition, you can fix the version of JaCoCo by adding this section into build.gradle
+```
+plugins {
+    // plugins area
+}
+jacoco {
+    toolVersion = "0.8.1"
+}
+```
+6. Moreover, there are other types of reports than just .html (but just .html is enough for this workshop).
+
+```
+plugins {
+    // plugins area
+}
+jacoco {
+    // jacoco's version
+}
+
+jacocoTestReport {
+    reports {
+        xml.enabled true
+        csv.enabled true
+    }
+}
+```
+
+## What have we done so far ?
+Up until now, we are able to analyze our code base using the service provide by SonarCloud. However, the analyze process is not yet automated. We are back to the same issue that we used to have with unit tests early in the beginning of this workshop. Every time we have made some changes into our code, we will have to use SonarCloud's analyze command and manually run it by ourselves. I'm sure you would not want to do it. Most of us might not even want to remember the command itself !
+
+## Integrate SonarCould with GitHub and Continuous Integration
+
+The objective for this section is to block the pull request that does not pass SonarCloud's standard from merging into the master branch. So the owners of the pull requests know that their PR need to be taken care of.
+
+### Use TravisCI to trigger SonarCloud's analysis
+1. Define SONAR_TOKEN in TravisCI repository's environment variable settings
+    1. Go to https://travis-ci.org/ and click at "ci-hands-on-workshop".
+
+    2. Click setting from the dropdown menu.
+    ![travis setting](https://user-images.githubusercontent.com/11821799/46256285-c9a4d800-c4d2-11e8-95e9-eb1468de3138.png)
+
+    3. Add your SonarCloud's token in Environment variable section. For simplicity of the workshop, let's give the token the same name, SONAR_TOKEN.
+    ![travis add token](https://user-images.githubusercontent.com/11821799/46256317-361fd700-c4d3-11e8-99da-303fcdfefb82.png)
+
+2. Get SonarCloud.io organization key
+    1. Visit this https://sonarcloud.io/account/organizations
+    
+    2. You should see the organization key near by your project.
+    ![sonar org key](https://user-images.githubusercontent.com/11821799/46256416-da564d80-c4d4-11e8-8c82-b8076133ce55.png)
+
+3. Create sonar-project.properties
```

Comment:
Just create an empty file ?

Reply:
my mistake, we dont really need this file. deleted this line 👍